openfermioncirq/gates/common_gates.py

```diff
@@ -253,3 +253,49 @@ def rot11(rads: float):
 FSWAP = FSwapPowGate()
 XXYY = XXYYPowGate()
 YXXY = YXXYPowGate()
+
+class CombinedSwapAndZ(
+        cirq.EigenGate,
+        cirq.InterchangeableQubitsGate,
+        cirq.TwoQubitGate):
+    """w0 * (XX + YY) + w1 * |11><11| interaction.
+
+    Equivalent to XXYYPowGate and CZ (in either order).
```

Comment:
I think this gate should be defined in terms of the XXYY and CZ _angles_ rather than weights. Then, we can add a helper function called `fermionic_simulation_gate` or something like that (taking the name from https://arxiv.org/pdf/1711.04789.pdf) which takes the weights. This approach is more consistent with what we do at Cirq (e.g. ZPowGate and Rz).

Reply:
But the weights here are effectively the same as the exponents. The ZPowGate/RZ thing is because there are different conventions for their parameterization, but that doesn't seem relevant here.

Reply:
I see, you're right. This still needs a better name though. I suggest FermionicSimulationGate.

Reply:
I changed to QuadraticFermionicSimulationGate and CubicFermionicSimulationGate.

What do you think of changing DoubleExcitationGate to QuarticFermionicSimulationGate?